src/transformer/descriptor/typeReference/typeReference.ts

```diff
@@ -9,6 +10,10 @@ import { GetTypescriptType, IsTypescriptType } from '../tsLibs/typecriptLibs';
 export function GetTypeReferenceDescriptor(node: ts.TypeReferenceNode, scope: Scope): ts.Expression {
     const declaration: ts.Declaration = TypescriptHelper.GetDeclarationFromNode(node.typeName);
 
+    if (MockDefiner.instance.hasMockForDeclaration(declaration)) {
```

Comment:
Considering that now we check up front if we already have the mock shall we simplify the checks inside GetMockFactoryCall? I think there are unnecessary check in this scenario. If you look at this line 

https://github.com/Typescript-TDD/ts-auto-mock/blob/323b697d2ded285d39d6d61b4cf46600092bbbf2/src/transformer/mockDefiner/mockDefiner.ts#L181

we will check the declaration in the cache twice.

In the same way the last if statement that use GetMockFactory call could be simplified to not to the check because it will add a new declaration in the cache and create a new mock

Reply:
I've changed few things:
- now in mockFactoryCall.ts there are 2 different ways to get the call, one where you create a new one, one where you use an existing one, this makes sure a check done earlier is not repeated
- in typeReference.ts we make the check and decide which function to use
- in mockDefiner the ifs are removed

The only downside of this is that now there is no check anymore, a call to "GetMockFactoryCall" of an existing mock factory will create a call to the repository of an existing factory. I would say that is ok, everything is under test so it shouldn't be possible to miss it.